pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -993,17 +1020,42 @@ class Timedelta(_Timedelta):
         return self.value // other.value
 
     def __rfloordiv__(self, other):
+        # numpy does not implement floordiv for timedelta64 dtype, so we cannot
+        # just defer
+        if hasattr(other, '_typ'):
+            # Series, DataFrame, ...
+            return NotImplemented
+
         if hasattr(other, 'dtype'):
-            # work with i8
-            other = other.astype('m8[ns]').astype('i8')
-            return other // self.value
+            if other.dtype.kind == 'm':
+                # also timedelta-like
```

Comment:
iirc for division we defer to. TDI to handle his logic
does this code path follow?

Reply:
> iirc for division we defer to. TDI to handle his logic

I would find that design choice surprising.

```
    def __truediv__(self, other):
        if hasattr(other, 'dtype'):
            return self.to_timedelta64() / other

        elif is_integer_object(other) or is_float_object(other):
            # integers or floats
            return Timedelta(self.value / other, unit='ns')

        elif not _validate_ops_compat(other):
            return NotImplemented

        other = Timedelta(other)
        if other is NaT:
            return np.nan
        return self.value / float(other.value)
```

I think the relevant difference here is for cases where `other` is an array this returns `NotImplemented`, but that isn't a viable option for floordiv because numpy doesn't implement floordiv for timedelta64.